pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyConfiguration.java

```diff
@@ -581,6 +581,12 @@
     )
     private Set<String> proxyExtensions = new TreeSet<>();
 
+    @FieldContext(
+            category = CATEGORY_PLUGIN,
+            doc = "Use a separate ThreadPool for each Proxy Extension"
+    )
+    private boolean useSeparateThreadPoolForProxyExtensions = true;
```

Comment:
It is shared now, should the default value be false?
In addition, do  we need to add unit tests?

Reply:
I want to set it to true by default because it is safer for the users.

if you are sharing the thread pool then you can reach to a situation in which the Broker is stuck and you do not notice it.

I could add a test case,
btw I think that no one will ever set this back to false, initially I thought to not add a configuration flag at all